src/testReportData.ts

```diff
@@ -0,0 +1,87 @@
+import { type TestReportConfig, testReportConfigSchema } from './testReportConfig.ts';
+import { type LcovFile, lcovParser, type LcovSummary } from './utilities/lcov_parser.ts';
+import { createJUnitParser, type TestSuites } from './utilities/junit_parser.ts';
+import { readTextFile } from './utilities/miscUtils.ts';
+
+async function loadReportConfig(reportDefinitionFilename: string): Promise<TestReportConfig> {
+  const reportConfigText = await readTextFile(reportDefinitionFilename);
+  return testReportConfigSchema.parse(JSON.parse(reportConfigText));
+}
+
+async function loadLcovData(lcovFilenames: string[]): Promise<LcovFile[]> {
+  const lcovDatas: LcovFile[] = [];
+  for (const lcovFilename of lcovFilenames) {
+    try {
+      const lcovData = await readTextFile(lcovFilename);
+      lcovDatas.push(...lcovParser(lcovData));
+    } catch (error) {
+      console.error((error as Error).message);
```

Comment:
return load errors as separate object? Note: files might not be created if test itself has failed